channels/applicationinsights-channel-js/Tests/Unit/src/Sender.tests.ts

```diff
@@ -48,13 +66,506 @@ export class SenderTests extends AITestClass {
+                QUnit.assert.deepEqual(arrBufferCopy.getItems(), [], "payload should be empty");
+                let payload = ["payload1", "payload2", "payload3", "payload4", "payload5", "payload6"];
+                arrForEach(payload, (val) =>{
+                    arrBuffer.enqueue(val);
+                });
+                arrBuffer.deepCopy(arrBufferCopy);
+                QUnit.assert.deepEqual(payload, arrBufferCopy.getItems(), "payload should be same");
+                arrBuffer.enqueue("payload");
+                QUnit.assert.deepEqual(arrBuffer.getItems().length, 7, "arrBuffer length");
+                QUnit.assert.deepEqual(arrBufferCopy.getItems().length, 6, "copy is deep copy");
+            }
+        });
+
+        this.testCase({
+            name: "ArraySendBuffer deepCopy: function deepCopy() can return expected sessionStorage buffer",
+            test: () => {
+                let config = {
+                    endpointUrl: "https//: test",
+                    emitLineDelimitedJson: false,
+                    maxBatchInterval: 15000,
+                    maxBatchSizeInBytes: 102400,
+                    disableTelemetry: false,
+                    enableSessionStorageBuffer: true,
+                    isRetryDisabled: false,
+                    isBeaconApiDisabled:true,
+                    disableXhr: false,
+                    onunloadDisableFetch: false,
+                    onunloadDisableBeacon: false,
+                    instrumentationKey:"key",
+                    namePrefix: "",
+                    samplingPercentage: 100,
+                    customHeaders: [{header:"header",value:"val" }],
+                    convertUndefined: "",
+                    eventsLimitInMem: 10000
+                } as ISenderConfig;
+                let logger = new DiagnosticLogger({instrumentationKey: "abc"});
+
+                let arrBuffer = new ArraySendBuffer(logger, config);
+                let sessionBuffer = new SessionStorageSendBuffer(logger, config);
+                arrBuffer.deepCopy(sessionBuffer);
+                QUnit.assert.deepEqual(sessionBuffer.getItems(), [], "payload should be empty");
+                let payload = ["payload1", "payload2", "payload3", "payload4", "payload5", "payload6"];
+                arrForEach(payload, (val) =>{
+                    arrBuffer.enqueue(val);
+                });
+                arrBuffer.deepCopy(sessionBuffer);
+                QUnit.assert.deepEqual( sessionBuffer.getItems(), payload, "payload should be same");
+                QUnit.assert.deepEqual(this._getBuffer(BUFFER_KEY, logger), payload, "session storage buffer is set");
+                arrBuffer.enqueue("payload");
+                QUnit.assert.deepEqual(arrBuffer.getItems().length, 7, "arrBuffer length");
+                QUnit.assert.deepEqual( sessionBuffer.getItems().length, 6, "copy is deep copy");
+
+                utlSetSessionStorage(logger, BUFFER_KEY,JSON.stringify([]));
```

Comment:
This should also check that the SessionStorage was updated and contains the expected values